docs/spec/auth/vesting.md

```diff
@@ -0,0 +1,98 @@
+## Vesting
+
+### Intro and Requirements
+
+This paper specifies changes to the auth and bank modules to implement vested accounts for the Cosmos Hub. 
+The requirements for this vested account is that it should be capable of being initialized during genesis with
+a starting balance X coins and a vesting blocknumber N. The owner of this account should be able to delegate to validators and vote,
+however they cannot send their coins to other accounts until the account has fully vested. Thus, the bank module's `MsgSend` handler 
+should error if a vested account is trying to send an amount before block N.
+
+### Implementation
+
+##### Changes to x/auth Module
+
+The `Account` interface will specify both the Account type and any parameters it needs.
+
+```go
+// Account is a standard account using a sequence number for replay protection
+// and a pubkey for authentication.
+type Account interface {
+    Type() string // returns the type of the account
+
+	GetAddress() sdk.AccAddress
+	SetAddress(sdk.AccAddress) error // errors if already set.
+
+	GetPubKey() crypto.PubKey // can return nil.
+	SetPubKey(crypto.PubKey) error
+
+	GetAccountNumber() int64
+	SetAccountNumber(int64) error
+
+	GetSequence() int64
+	SetSequence(int64) error
+
+	GetCoins() sdk.Coins
+    SetCoins(sdk.Coins) error
+    
+    // Getter and setter methods for account params
+    // It is upto handler to use these appropriately
+    GetParams() map[string]interface{}
+    SetParams(map[string]interface{}) error
+}
+```
+
+The `Type` method will allow handlers to determine what type of account is sending the message, and the 
+handler can then call `GetParams` to handle the specific account type using the parameters it expects to 
+exist in the parameter map.
+
+The `VestedAccount` will be an implementation of `Account` interface that wraps `BaseAccount` with 
+`Type() => "vested` and params, `GetParams() => {"BlockLock": blockN (int64)}`. 
+`SetParams` will be disabled as we do not want to update params after vested account initialization.
```

Comment:
I'm wondering if we can just eliminate `Type` (are we introducing it?) and only have a block-lock param - if it exists then we know it's vested, or rather just locked - which is the important part

Reply:
Yea I want to introduce it. If we just want to support this specific case, we can. I think we should add the Type tho, because we may want to add other types of Account. And people building on top of sdk should be able to create multiple account types as well.

For example, in the future we may want to support NLocked Accounts where each month a new amount of coins get unlocked, or maybe some other feature. I don't think we should add cruft to BaseAccount every time we want something new and specific to our application.

imo, it makes sense to create different Account types that define their own parameters and have handlers that can handle specific account types.

Will move a lot of this into a proposal issue as you suggested and rewrite.
Thanks for the review!